src/dbos-runtime/config.ts

```diff
@@ -481,9 +481,11 @@ export function overwrite_config(
 
   const poolConfig = constructPoolConfig(configFile!);
 
-  const OTLPExporterConfig: OTLPExporterConfig = providedDBOSConfig.telemetry?.OTLPExporter || {};
+  if (!providedDBOSConfig.telemetry.OTLPExporter) {
+    providedDBOSConfig.telemetry.OTLPExporter = {};
+  }
   if (configFile!.telemetry?.OTLPExporter?.tracesEndpoint) {
-    OTLPExporterConfig.tracesEndpoint = configFile!.telemetry.OTLPExporter.tracesEndpoint;
+    providedDBOSConfig.telemetry.OTLPExporter.tracesEndpoint = configFile!.telemetry.OTLPExporter.tracesEndpoint;
```

Comment:
The previous code wasn't setting the overwrite to a reference of `providedDBOSConfig` if it didn't have an OTLP config